cmd/metal-api/internal/service/partition-service.go

```diff
@@ -359,9 +358,11 @@ func (r partitionResource) calcPartitionCapacity() ([]v1.PartitionCapacity, erro
 
 			cap.Total++
 		}
-
 		sc := []v1.ServerCapacity{}
 		for i := range capacities {
+			if capacities[i] == nil {
```

Comment:
Oh, interesting! This seems to be fixing a bug but also seems unrelated to this PR. Can we file this in another issue and a separate PR? :)